pkgs/development/compilers/openjdk/jre.nix

```diff
@@ -1,19 +1,34 @@
-{ jdk
-, runCommand
-, patchelf
+{ stdenv
+, jdk
 , lib
 , modules ? [ "java.base" ]
 }:
 
 let
-  jre = runCommand "${jdk.name}-jre" {
-    nativeBuildInputs = [ patchelf ];
+  jre = stdenv.mkDerivation {
+    name = "${jdk.name}-minimal-jre";
+    version = jdk.version;
+
     buildInputs = [ jdk ];
+
+    dontUnpack = true;
+
+    # Strip more heavily than the default '-S', since if you're
+    # using this derivation you probably care about this.
+    stripDebugFlags = [ "--strip-unneeded" ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      jlink --module-path ${jdk}/lib/openjdk/jmods --add-modules ${lib.concatStringsSep "," modules} --output $out
+
+      runHook postBuild
+    '';
+
+    dontInstall = true;
+
     passthru = {
       home = "${jre}";
     };
-  }   ''
-      jlink --module-path ${jdk}/lib/openjdk/jmods --add-modules ${lib.concatStringsSep "," modules} --output $out
-      patchelf --shrink-rpath $out/bin/* $out/lib/jexec $out/lib/jspawnhelper $out/lib/*.so $out/lib/*/*.so
```

Comment:
I suppose `--shrink-rpath` has become redundant?
It would be nice to have a test in `passthru`. Perhaps just a small package that uses this jre in its tests.

Reply:
> I suppose `--shrink-rpath` has become redundant?

Yes - I tried `patchelf --shrink-rpath` on the resulting binaries and it doesn't produce any change anymore

> It would be nice to have a test in `passthru`. Perhaps just a small package that uses this jre in its tests.

I agree that would be nice, though I'd rather not grow this PR with it..

Reply:
Yeah I respect that.